google/cloud/storage/internal/object_requests.cc

```diff
@@ -408,16 +408,25 @@ std::string UploadChunkRequest::RangeHeader() const {
     // the range is special in this case.
     os << "*";
   } else {
-    os << range_begin() << "-" << range_begin() + size - 1;
+    os << offset() << "-" << offset() + size - 1;
   }
-  if (!last_chunk_) {
+  if (!upload_size().has_value()) {
     os << "/*";
   } else {
-    os << "/" << source_size();
+    os << "/" << *upload_size();
   }
   return std::move(os).str();
 }
 
+UploadChunkRequest UploadChunkRequest::RemainingChunk(
+    std::uint64_t new_offset) const {
+  UploadChunkRequest result = *this;
+  if (new_offset < offset()) return result;
```

Comment:
Opt: Would it be clearer to use `result.offset_` there?

Reply:
Fixed.